packages/@react-aria/focus/test/FocusScope.test.js

```diff
@@ -285,6 +285,40 @@ describe('FocusScope', function () {
 
       expect(document.activeElement).toBe(outside);
     });
+
+    it('should not restore focus back to the previous node if a new FocusScope becomes the active scope', function () {
+      function Test({show}) {
+        return (
+          <div>
+            <input data-testid="outside" />
+            <FocusScope restoreFocus contain>
+              <input data-testid="input1" />
+            </FocusScope>
+            {show &&
+              <FocusScope restoreFocus contain isActiveScope>
+                <input data-testid="input3" />
+              </FocusScope>
+            }
+          </div>
+        );
+      }
+
+      let {getByTestId, rerender} = render(<Test />);
+      // Set a focused node and make first FocusScope the active scope
+      let input1 = getByTestId('input1');
+      input1.focus();
+      fireEvent.focusIn(input1);
+      expect(document.activeElement).toBe(input1);
+      
+      // Rerender with a new active FocusScope
+      rerender(<Test show />);
+
+      expect(document.activeElement).toBe(input1);
+      let input3 = getByTestId('input3');
+      input3.focus();
+      fireEvent.focusIn(input3);
+      expect(document.activeElement).toBe(input3);
+    });      
```

Comment:
Lemme know if this new test makes sense. It does the following:
1. Renders a single FocusScope and focuses the input inside it. This is to get an `activeScope` set and a item saved for `focusedNode.current`
2. Rerenders a new FocusScope that is set as the new `activeScope` via `isActiveScope`.
3. Attempts to focus an item inside it. Since the new FocusScope is the `activeScope` and the item is in the scope, FocusScope doesn't reset focus to input1.

Tested the negative case locally by removing the `isActiveScope` and confirmed that the above test fails and input1 was still focused (aka the current failing dialog test case)